### PR TITLE
Use --config ... and Config.defaultConfigPath for test-safe configuration. Fixes gh-38

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+      value: configPath
+    });
     this.load();
   }
 
   load() {
     this.hosts = {}
 
-    if (!fs.existsSync(this.configPath)) return;
+    if (!fs.existsSync(this.configPath)) {
```
Comment (bterlson) [inline lib/config.js]: FWIW I think the one-line style is perfectly acceptable for control flow statements (pretend it's a return-if special form ;)) but I understand prettier/etc. auto-fixes this so meh.

File: test/bin/eshost.js [hunk 2]
```diff
@@ -26,6 +26,8 @@ if (isWindows) {
   });
 }
 
+// To avoid messing with a real configuration, use a test-local config
+Config.defaultConfigPath = path.resolve('./test/config/.eshost-config.json');
```
Comment (bterlson) [inline test/bin/eshost.js]: Is this necessary? Seems like the --config option takes care of this.

Another option is to use an environment variable to control where the defaultConfigPath is. Then you don't need to pass it in.

Comment (rwaldron) [inline test/bin/eshost.js]: Oh, yes, I think this is unnecessary now. 

> Then you don't need to pass it in.

I determined that passing it in explicitly acts as a reasonable test of `--config` option :)

File: lib/config.js (hunk 1 above)
Comment (rwaldron) [inline lib/config.js]: Fair enough.

**Comment by rwaldron** [comment]
If #44 lands first, I will rebase this branch and push up so we can see it go through appveyor!

**Pull request merged**